script/lib/filesystem.py

```diff
@@ -51,15 +51,18 @@ def download_and_extract(destination, url):
     with contextlib.closing(urllib2.urlopen(url)) as u:
       total = int(u.headers['content-length'])
       done = 0
+      last_length = 0
       while True:
         chunk = u.read(1024)
         done += len(chunk)
         if not len(chunk):
           break
-        sys.stderr.write('                                   \r')
-        percent = str(round(float(done) / float(total), 4) * 100) + '%'
+        percent = '{0:.2f}'.format(round(float(done) / float(total), 4) * 100) + '%'
```

Comment:
(minor) I think you could avoid joining two strings by making the first string `'{0:.2f}%%'`